packages/lib/gpcircuits/circuits/bounds.circom

```diff
@@ -0,0 +1,67 @@
+pragma circom 2.1.8;
+
+include "circomlib/circuits/comparators.circom";
+include "circomlib/circuits/bitify.circom";
+include "gpc-util.circom";
+
+/**
+ * Module carrying out bounds checks on 64-bit unsigned integer values.
+ * 
+ * No overall enable flag, but can be disabled by checking whether 0
+ * lies in the interval [0, 1 << 64 - 1].
+ * 
+ * The values and bounds are assumed to be 64-bit unsigned integers
+ * but are not constrained here, cf. the comments regarding {@link
+ * InInterval}.
+ */
+template BoundsCheckModule(
+    // Number of values to be checked.
+    MAX_BOUNDS_CHECKS
+) {
+    // Values to be checked.
+    signal input comparisonValues[MAX_BOUNDS_CHECKS];
+
+    // Bounds that each corresponding value should lie within.
+    signal input bounds[MAX_BOUNDS_CHECKS][2];
+
+    // Indicator of whether the bounds check was successful.
+    signal boundsCheck[MAX_BOUNDS_CHECKS];
+
+    // Loop through and check each value against its expected bounds.
+    for(var i = 0; i < MAX_BOUNDS_CHECKS; i++) {
+        boundsCheck[i] <== InInterval(64)(comparisonValues[i], bounds[i]);
+        boundsCheck[i] === 1;
+    }
+}
+
+/**
+ * Module checking whether a single `NUM_BITS`-bit unsigned integer
+ * lies in a closed interval with `NUM_BITS`-bit unsigned integer
+ * bounds. Outputs a boolean.
+ *
+ * Signals are assumed to be in the interval [0, 1 << NUM_BITS[, but
```

Comment:
I assume the reversed square-bracket is a typo?  Or is that a way of expressing a half-open interval I haven't seen before?
This should either be a half-open interval like `[0, 1<< NUM_BITS)`, or it needs a -1 `[0, (1<<NUM_BITS)-1]`.